### PR TITLE
10.0 fix ir attachment url avatar

### PR DESCRIPTION
**Pull request by iledarn**

### CODE REVIEW

File: ir_attachment_url/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from odoo.addons.web.controllers.main import binary_content
+from odoo.exceptions import AccessError
+
+from odoo.addons.mail.controllers.main import MailController
+from ..models.image import is_url
+
+
+class MailControllerExtended(MailController):
+
+    @http.route('/mail/<string:res_model>/<int:res_id>/avatar/<int:partner_id>', type='http', auth='public')
```
Comment (yelizariev) [inline ir_attachment_url/controllers/main.py]: Use empty decorator if you don't change anything there, i.e.

``@http.route()``

File: ir_attachment_url/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                pass
+
+        if status == 301 and is_url(content):
+            r = requests.get(content)
+            image_base64 = r.content
+            headers.append(('Content-Length', len(image_base64)))
+            response = request.make_response(image_base64, headers)
+            response.status = str(status)
+        else:
+            response = super(MailControllerExtended, self).avatar(res_model, res_id, partner_id)
```
Comment (yelizariev) [inline ir_attachment_url/controllers/main.py]: It's probably better to don't call super to repeat access checking and so on

Comment (iledarn) [reply ir_attachment_url/controllers/main.py]: I don't think that not using the super is better than repeat some code

Comment (iledarn) [reply ir_attachment_url/controllers/main.py]: The other way may be to take the super `response` and check if response.status == '301' and then try to decode again response.data - to get the url and then get the url's data and put it in response.data

Comment (iledarn) [reply ir_attachment_url/controllers/main.py]: The other way is failed on encoding decoded url

>>> import base64
>>> url = 'https://www.it-projects.info'
>>> urldecoded = base64.b64decode(url)
>>> urldecoded
'\x86\xdbi\xb3\xff\xf0\xc3\x08\xad\xa6\xba#y\xcbl\x8aw\xe8'
>>> urlafter = base64.b64encode(urldecoded)
>>> urlafter
'https//wwwitprojectsinfo'

**Comment by ilmir-k** [comment]
@iledarn need to change module version

File: ir_attachment_url/__init__.py [hunk 3]
```diff
@@ -5,3 +5,4 @@
 def post_load():
     from .models import binary_fields
     from .models import image
+    from . import controllers
```
Comment (yelizariev) [inline ir_attachment_url/__init__.py]: Add comment Why

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Pull request merged**